src/server/jobs/prepare-bounties.ts

```diff
@@ -7,28 +7,158 @@ import { createBuzzTransaction } from '~/server/services/buzz.service';
 import { TransactionType } from '~/server/schema/buzz.schema';
 import { Tracker } from '../clickhouse/client';
 import { handleTrackError } from '../utils/errorHandling';
+import {
+  bountyAutomaticallyAwardedEmail,
+  bountyExpiredEmail,
+  bountyExpiredReminderEmail,
+  bountyRefundedEmail,
+} from '~/server/email/templates';
 
-const log = createLogger('bounties', 'blue');
+const log = createLogger('prepare-bounties', 'blue');
 
-const prepareBounties = createJob('prepare-bounties', '0 1 * * *', async () => {
+const prepareBounties = createJob('prepare-bounties', '0 23 * * *', async () => {
   const [lastRun, setLastRun] = await getJobDate('prepare-bounties');
+  const justExpiredBounties = await dbWrite.bounty.findMany({
+    where: {
+      complete: false,
+      // Expires today
+      expiresAt: dayjs().toDate(),
+      userId: { not: null },
+      user: {
+        email: { not: null },
+      },
+      entries: {
+        some: {},
+      },
+    },
+    select: {
+      id: true,
+      name: true,
+      userId: true,
+      user: {
+        select: {
+          id: true,
+          email: true,
+        },
+      },
+      _count: {
+        select: {
+          entries: true,
+        },
+      },
+    },
+  });
 
-  const bounties = await dbWrite.bounty.findMany({
+  log(
+    'justExpiredBounties IDs',
+    justExpiredBounties.map((b) => b.id)
+  );
+
+  // send emails to just expired bounties:
+  for (const { id, userId, user, name, _count } of justExpiredBounties) {
+    log('Sending bounty expired reminder to ', userId);
+    if (user?.email) {
+      await bountyExpiredEmail.send({
+        bounty: {
+          id,
+          name,
+          entryCount: _count.entries ?? 0,
+        },
+        user: {
+          email: user.email,
+        },
+      });
+    }
+  }
```

Comment:
This would hold the job until it foes through all expired bounties sequentially, do you think it could be possible to do them concurrently with a `Promise.all`? Or just let them finish in the background

Reply:
I'll let them finish on the BG most likely then 👍